application/libraries/REST_Controller.php

```diff
@@ -1028,21 +1028,26 @@ protected function _detect_lang()
      */
     protected function _log_request($authorized = FALSE)
     {
+        
+        $insert_id = $this->rest->db->select_max("ID","insert_id")->get($this->config->item('rest_logs_table'))->row(); //Gets last ID because not all databases support IDENTITY columns (Like Oracle)
+        $insert_id = (count($insert_id)>0) ? $insert_id->insert_id + 1 : 1; //if there's no rows in the table there'll be a null result then we set the first id to be 1, otherwise it'll be the greatest one plus 1
```

Comment:
Kept this line appart because function chaining with all of this would make it difficult to read